### ISSUE TITLE
Access a Vpc's Internet Gateway

### ISSUE THREAD

**Issue by jeanblanchard** [aws/aws-cdk#5327]
Add an accessor to the Internet Gateway that was created by the `Vpc` construct.

### Use Case

Trying to add a `PublicSubnet` to an existing `Vpc` (with existing public subnets), there is no way to create a route to the `Vpc`'s existing Internet Gateway, because the reference is not exposed anywhere (that I could find), so the `PublicSubnet` is not effectively public.

### Proposed Solution

* Either add an accessor to a VPC's internet gateway, if any
* Or have the `PublicSubnet` constructor automagically create the route from the VPC it's a part of.

(Or, ideally, both)

### Other

I'd take a workaround within CDK, that allows this without creating (and paying for) a second IGW...

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by jeanblanchard** [comment]
Update - I found my workaround, using an [escape hatch](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html) to get the "IGW" child resource in the `Vpc` construct:
```ts
    const igw = vpc.node.findChild('IGW') as CfnInternetGateway;
    new CfnRoute(this, 'IGW', {
      routeTableId: subnet.routeTable.routeTableId,
      destinationCidrBlock: '0.0.0.0/0',
      gatewayId: igw.ref,
    });
```

**Comment by SomayaB** [comment]
Glad you could figure it out! :) Closing this issue. Feel free to reopen.

**Comment by shearn89** [comment]
Having a stab at this, hope to have a PR out soon.

**Comment by deepu-james** [comment]
> ```ts
> const igw = vpc.node.findChild('IGW') as CfnInternetGateway;
> ```

@jeanblanchard  This is not working now. 
`const igw = existingDefaultVPC.node.findChild('IGW') as ec2.CfnInternetGateway;` throws an error message "No child with id: 'IGW' at Node.findChild"
Is there a different way to get the internet gateway id from an imported VPC?

**Comment by daverickdunn** [comment]
Don't know if this is a regression, or if it was never fixed, but this still doesn't work in `1.134.0` and it previously didn't work in `1.117.0`. The `internetGatewayId` field of `Vpc` is `undefined` when using `Vpc.fromLookup`. Same applies for the proposed workaround above.

**Comment by shearn89** [comment]
Hi @daverickdunn - unless it was added since I did the work on this then the PR never supported getting the IGW ID from an imported VPC, as stated in the [README](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-ec2#accessing-the-internet-gateway): I just didn't have the time to add it in since that wasn't my use case!

Still, I'm curious why the escape hatch isn't working, unless the child name has changed. I'll see if I can dig through and find out.
